finetune/encoding/target_encoders.py

```diff
@@ -181,12 +185,84 @@ def target_labels(self):
         raise ValueError
 
 
-class SequenceLabelingEncoder(LabelEncoder, BaseEncoder):
-    pass
+class SequenceLabelingEncoder(BaseEncoder):
+
+    def __init__(self, pad_token="<PAD>"):
+        self.classes_ = None
+        self.pad_token = pad_token
+        self.lookup = None
+
+    def fit(self, labels):
+        self.classes_ = list(set(lab["label"] for lab in labels) | {self.pad_token})
+        self.lookup = {c: i for i, c in enumerate(self.classes_)}
+
+    def pre_process_label(self, out, labels):
+        # TODO employ more complex strategies
+        pad_idx = self.lookup[self.pad_token]
+        return labels, pad_idx
+
+    @staticmethod
+    def overlaps(label, tok_start, tok_end, tok_text):
+        does_overlap = (
+            label["start"] < tok_end <= label["end"] or
+            tok_start < label["end"] <= tok_end
+        )
+        if not does_overlap:
+            return False, False
+
+        start = max(tok_start, label["start"])
+        end = min(tok_end, label["end"])
+        sub_text = label["text"][start - label["start"]: end - label["end"]]
+        print(sub_text)
+        print(tok_text)
+        strings_agree = sub_text.lower() in tok_text.lower()
+        return does_overlap, strings_agree
+
+    def transform(self, out, labels):
+        labels, pad_idx = self.pre_process_label(out, labels)
+        labels_out = [pad_idx for _ in out.tokens]
+        for label in labels:
+            for i, (start, end, text) in enumerate(zip(out.token_starts, out.token_ends, out.tokens)):
```

Comment:
Can see this scaling poorly with sequence length.  Might be worth sorting + exiting early?